jwst/stpipe/tests/test_integration.py

```diff
@@ -18,3 +19,36 @@ def test_get_steps():
         assert step_class.class_alias == class_alias
         if is_pipeline:
             assert issubclass(step_class, Pipeline)
+
+
+def _get_subclass_names(class_, ignore=None, ignore_tests=True):
+    """
+    Iterate through names of all subclasses (recursively) of a class ignoring
+    all subclasses with names in ignore (and ignoring any subclasses of
+    ignored classes). Also optionally ignore all classes defined in tests modules.
+    """
+    if ignore is None:
+        ignore = set()
+    for subclass in class_.__subclasses__():
+        if subclass.__name__ in ignore:
+            continue
+        if ignore_tests and (subclass.__module__ == 'local' or 'tests' in subclass.__module__.split('.')):
+            continue
+        yield subclass.__name__
+        yield from _get_subclass_names(subclass, ignore, ignore_tests)
```

Comment:
There's something very similar to this function in

https://github.com/spacetelescope/jwst/blob/b762c669639416d61b6e7d3facc5cbee2d6bc9b0/jwst/lib/suffix.py#L261-L307

where independently the idea is to find all suffixes of all steps, but of course it doesn't take into account anything in `jwst.stpipe.integration`, which it should.  It also is only run during a test to verify the static list is up-to-date.  Better to have it in test infrastructure.

We may want refactor the suffix handling.  And I have a good reason to refactor suffix handling anyway so that it is more dynamic - handling suffixes of other non-`jwst` package steps that are registered through stpipe entry points, but doing so without having to import all the JwstStep and JwstPipeline subclasses to do so.  I will work on this.